client/app/account/billing/confirmTerminate/billing-confirmTerminate.controller.js

```diff
@@ -1,57 +1,81 @@
-            this.globalError = true;
-            return;
-        }
+        this.init = function () {
+            if (!this.token || !this.serviceId) {
+                this.globalError = true;
+                return;
+            }
 
-        this.loading = true;
-        BillingTerminate
-            .getServiceInfo(this.serviceId)
-            .then((serviceInfos) => {
-                this.serviceInfos = serviceInfos;
-                return serviceInfos.serviceId;
-            })
-            .then((serviceId) =>
-                BillingTerminate.getServiceApi(serviceId, true).then((service) => {
-                    this.serviceState = _.get(service, "resource.state");
-                })
-            )
-            .catch(() => {
+            $q.all([
+                this.loadUser(),
+                this.loadService()
+            ]).catch(() => {
                 this.globalError = true;
-            })
-            .finally(() => {
-                this.loading = false;
-            });
-    };
-
-    this.confirmTermination = function () {
-        this.terminating = true;
-        BillingTerminate
-            .confirmTermination(this.serviceId, this.serviceInfos.domain, this.reason, this.commentary, this.token)
-            .then(() => {
-                this.error = false;
-                this.serviceState = "suspending";
-            })
-            .catch(() => {
-                this.error = true;
-            })
-            .finally(() => {
-                this.terminating = false;
             });
-    };
+        };
+
+        this.loadUser = function () {
+            return User.getUser()
+                .then((user) => {
+                    this.USVersion = user && user.ovhSubsidiary === "US";
+                })
+                .finally(() => {
+                    this.userLoading = false;
+                });
+        };
+
+        this.loadService = function () {
+            return BillingTerminate
+                .getServiceInfo(this.serviceId)
+                .then((serviceInfos) => {
+                    this.serviceInfos = serviceInfos;
+                    return serviceInfos.serviceId;
+                })
+                .then((serviceId) => BillingTerminate.getServiceApi(serviceId, true).then((service) => {
+                    this.serviceState = _.get(service, "resource.state");
+                }))
+                .finally(() => {
+                    this.loading = false;
+                });
+        };
+
+        this.confirmTermination = function () {
+            this.terminating = true;
+            BillingTerminate
+                .confirmTermination(this.serviceId, this.serviceInfos.domain, this.reason, this.commentary, this.token)
+                .then(() => {
+                    this.error = false;
+                    this.serviceState = "suspending";
+                })
+                .catch(() => {
+                    this.error = true;
```

Comment:
You don't retrieve the error message ? Because a `504` is not the same error as a `Termination can't be terminated because...`

Reply:
The error is global and not declined for each error type so this is useless here.